views/xTree/XTreeView.js

```diff
@@ -0,0 +1,1323 @@
+    async displayAncestorList() {
+        //$(this.selector).html("Working....");
+        wtViewRegistry.showNotice(`Building the ancestor list to a max of ${this.maxGeneration} generations...`);
+
+        let data = await WikiTreeAPI.postToAPI({
+            appId: XTreeView.APP_ID,
+            action: "getPerson",
+            key: this.startId,
+            fields: this.profileFields,
+        });
+        if (data.length != 1) {
+            wtViewRegistry.showError(`There was an error starting with ${this.startId}.`);
+            return;
+        }
+
+        // Yay, we have a valid starting person.
+        // If the profile is private and the viewing user is not on the Trusted List, we still might not be able to continue.
+        let p = data[0].person;
+        XTreeView.PrimaryPerson = p;
+        if (!p?.Name) {
+            let err = `The starting profile data could not be retrieved.`;
+            if (wtViewRegistry?.session.lm.user.isLoggedIn()) {
+                err += ` You may need to be added to the starting profile's Trusted List.`;
+            } else {
+                err += ` Try the Apps Login.`;
+            }
+            wtViewRegistry.showError(err);
+            wtViewRegistry.hideInfoPanel();
+            return;
+        }
+        if (p.Privacy < 50 && !p.Gender) {
+            wtViewRegistry.showError(
+                `<p>Sorry, this profile is <a href="/wiki/Privacy">Private</a> and you are not on the profile's <a href="/wiki/Trusted_List">Trusted List</a>.</p>`
+            );
+            wtViewRegistry.hideInfoPanel();
+            return;
+        }
+
+        // Fill in some custom links in the "description" with completed values.
+        let x = p.Name.split("-");
+        let count = x[x.length - 1];
+        $("#familyListLink").attr("href", `https://www.wikitree.com/index.php?title=Special:FamilyList&p=${p.Id}`);
+        $("#compactTreeLink").attr("href", `https://www.wikitree.com/treewidget/${p.Name}`);
+        $("#toolsLink").attr("href", `https://www.wikitree.com/genealogy/${p.LastNameAtBirth}-Family-Tree-${count}`);
+        $("#toolsLink").html(`${p.RealName}'s Tree &amp; Tools page`);
+
+        // Display our "info" panel with a description of this view.
+        wtViewRegistry.showInfoPanel();
+
+        // Setup the Button Bar --> Initial version will use mostly text links, but should be replaced with icons - ideally images that have a highlighted / unhighlighted version, where appropriate
+        let btnBarHTML =
+            '<table border=0 style="background-color: #f8a51d80;" width="100%"><tr>' +
+            '<td width="30%">' +
+            "&nbsp;" +
+            "<input type=radio name=appMode checked value=Probability onclick='XTreeView.changeMode(0);'>Probability Tree&nbsp;&nbsp; <input type=radio name=appMode value=Random onclick='XTreeView.changeMode(1);'>Random Simulation&nbsp;&nbsp; "+
```

Comment:
better use `<label>`  element for inputs ([docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label)), it'll improve usability of the input ... click on the label triggers the input.